### PR TITLE
Update ProxyTarantoolClient documentation

### LINKED ISSUE #307: Update Proxy Tarantool client documentation
Update Proxy Tarantool client documentation (single router connection)

### PR DESCRIPTION
**Pull request by iDneprov**
Update ProxyTarantoolClient documentation according to our new documentation policy

I haven't forgotten about:
- [x] Tests
- [ ] Changelog
- [x] Documentation
- [x] Commit messages comply with the [guideline](https://www.tarantool.io/en/doc/latest/dev_guide/developer_guidelines/#how-to-write-a-commit-message)
- [x] Cleanup the code for review. See [checklist](https://github.com/tarantool/cartridge-java/blob/master/docs/review-checklist.md)

Related issues:

Closes #307

### CODE REVIEW

File: src/test/java/io/tarantool/driver/integration/ProxyTarantoolClientExampleIT.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            client.getConfig().getMessagePackMapper());
+
+        TarantoolTuple tuple;
+        List<CompletableFuture<?>> allFutures = new ArrayList<>(20);
+        for (int i = 0; i < 20; i++) {
+            // Pass the field corresponding to bucket_id as null for tarantool/crud to compute it automatically
+            tuple = tupleFactory.create(1_000_000 + i, null, "FIO", 50 + i, 100 + i);
+            allFutures.add(space.insert(tuple));
+        }
+        allFutures.forEach(CompletableFuture::join);
```
Comment (ArtDu) [inline src/test/java/io/tarantool/driver/integration/ProxyTarantoolClientExampleIT.java:76]: Maybe we should use crud.insert_batch

File: src/test/java/io/tarantool/driver/integration/ProxyTarantoolClientExampleIT.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        conditions = Conditions.greaterOrEquals(PK_FIELD_NAME, 1_000_000 + 10);
+        selectResult = space.select(conditions).get();
+        assertEquals(10, selectResult.size());
+
+        // space.delete can delete only a single tuple by primary index value
+        // Do not use conditions like after or greaterOrEquals
+        conditions = Conditions.equals(PK_FIELD_NAME, 1_000_000 + 15);
+        TarantoolResult<TarantoolTuple> deleteResult = space.delete(conditions).get();
+
+        assertEquals(1, deleteResult.size());
```
Comment (ArtDu) [inline src/test/java/io/tarantool/driver/integration/ProxyTarantoolClientExampleIT.java:104]: And you can add `update` here to complete CRUD acronym

**Review by ArtDu — CHANGES_REQUESTED** [review]

File: src/test/java/io/tarantool/driver/integration/ProxyTarantoolClientExampleIT.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            // Pass the field corresponding to bucket_id as null for tarantool/crud to compute it automatically
+            tarantoolTuples.add(tupleFactory.create(1_000_000 + i, null, "FIO", 50 + i, 100 + i));
+        }
+        space.insertMany(tarantoolTuples).get();
+
+        // Primary index key value will be determined from the tuple
+        tuple = tupleFactory.create(1_000_000 - 1, null, "FIO", 50 - 1, 100 - 1);
+        Conditions conditions = Conditions.after(tuple);
+
+        TarantoolResult<TarantoolTuple> selectResult = space.select(conditions).get();
```
Comment (ArtDu) [inline src/test/java/io/tarantool/driver/integration/ProxyTarantoolClientExampleIT.java:82]: nit: I don't know if it is more beautiful or not?
```suggestion
        TarantoolResult<TarantoolTuple> selectResult = space.select(
            Conditions.greaterOrEquals(PK_FIELD_NAME, 1_000_000)
        ).get();
```

File: src/test/java/io/tarantool/driver/integration/ProxyTarantoolClientExampleIT.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        selectResult = space.select(conditions).get();
+        assertEquals(10, selectResult.size());
+
+        // space.delete can delete only a single tuple by primary index value
+        // Do not use conditions like after or greaterOrEquals
+        conditions = Conditions.equals(PK_FIELD_NAME, 1_000_000 + 15);
+        TarantoolResult<TarantoolTuple> deleteResult = space.delete(conditions).get();
+        assertEquals(1, deleteResult.size());
+
+        conditions = Conditions.indexEquals(PK_FIELD_NAME, Collections.singletonList(1_000_000 + 10));
```
Comment (ArtDu) [inline src/test/java/io/tarantool/driver/integration/ProxyTarantoolClientExampleIT.java:105]: It should be work fine with this
```suggestion
        conditions = Conditions.indexEquals(PK_FIELD_NAME, 1_000_000 + 10);
```

**Review by ArtDu — CHANGES_REQUESTED** [review]

File: src/test/java/io/tarantool/driver/integration/ProxyTarantoolClientExampleIT.java (hunk 3 above)
Comment (iDneprov) [reply src/test/java/io/tarantool/driver/integration/ProxyTarantoolClientExampleIT.java:82]: I've tried to show multiple filter options

**Review by ArtDu — APPROVED** [review]

**Pull request merged**